mmv1/products/edgecontainer/Cluster.yaml

```diff
@@ -0,0 +1,492 @@
+          after creation.
+      - !ruby/object:Api::Type::Array
+        name: "clusterIpv6CidrBlocks"
+        item_type: Api::Type::String
+        immutable: true
+        description: |
+          If specified, dual stack mode is enabled and all pods in the cluster are
+          assigned an IPv6 address from these blocks alongside from an IPv4
+          address. Only a single block is supported. This field cannot be changed
+          after creation.
+      - !ruby/object:Api::Type::Array
+        name: "servicesIpv6CidrBlocks"
+        item_type: Api::Type::String
+        immutable: true
+        description: |
+          If specified, dual stack mode is enabled and all services in the cluster are
+          assigned an IPv6 address from these blocks alongside from an IPv4
+          address. Only a single block is supported. This field cannot be changed
+          after creation.
+      - !ruby/object:Api::Type::Enum
+        name: "networkType"
+        description: |
+          IP addressing type of this cluster i.e. SINGLESTACK_V4 vs DUALSTACK_V4_V6.
+        values:
+          - :NETWORK_TYPE_UNSPECIFIED
+          - :SINGLESTACK_V4
+          - :DUALSTACK_V4_V6
+  - !ruby/object:Api::Type::NestedObject
+    name: "authorization"
+    required: true
+    immutable: true
+    description: |
+      RBAC policy that will be applied and managed by GEC.
+    properties:
+      - !ruby/object:Api::Type::NestedObject
+        name: "adminUsers"
+        required: true
+        description: |
+          User that will be granted the cluster-admin role on the cluster, providing
+          full access to the cluster. Currently, this is a singular field, but will
+          be expanded to allow multiple admins in the future.
+        properties:
+          - !ruby/object:Api::Type::String
+            name: "username"
+            description: |
+              An active Google username.
+            required: true
+  - !ruby/object:Api::Type::Integer
+    name: "defaultMaxPodsPerNode"
+    default_from_api: true
+    description: |
+      The default maximum number of pods per node used if a maximum value is not
+      specified explicitly for a node pool in this cluster. If unspecified, the
+      Kubernetes default value will be used.
+  - !ruby/object:Api::Type::String
+    name: "endpoint"
+    description: |
+      The IP address of the Kubernetes API server.
+    output: true
+  - !ruby/object:Api::Type::Integer
+    name: "port"
+    description: |
+      The port number of the Kubernetes API server.
+    output: true
+  - !ruby/object:Api::Type::String
+    name: "clusterCaCertificate"
+    description: |
+      The PEM-encoded public certificate of the cluster's CA.
+    output: true
+    sensitive: true
+  - !ruby/object:Api::Type::NestedObject
+    name: "maintenancePolicy"
+    description: |
+      Cluster-wide maintenance policy configuration.
+    properties:
+      - !ruby/object:Api::Type::NestedObject
+        name: "window"
```

Comment:
I think some of these nested fields should be `required: true` in the context of their parent, unless there is meaning when they are not specified but the parent is, ie. `maintenance_policy: {}`

Reply:
The maintenancePolicy is an optional field, but if the maintenancePolicy is provided the window need to provided. 

But I am wondering will the `required: true` on the children filed got ignored if the parent filed is unset?

Reply:
Yes, the required attribute applies only within the parent

Reply:
Thanks, added required: true into the children fields of maintenancePolicy.